Renamed constants to prevent common variable names being reserved

thareh opened the pull request.

thareh committed:
Renamed constants

woollybah commented:
Rather than a simple name change, it'd probably be better to use Enums instead...
```
Enum EMonth
 January = 1
 February
 ...


Enum EWeekday
 Sunday
 Monday
...
```

thareh commented:
You are completely right, I haven't really gotten in to the habit of using enums. I'll update the PR.

thareh committed:
Replaced consts with enums

Added overloads for all functions/methods (I think), this is the way to do it yes? (Haven't used Enums before)

thareh commented:
Replaced the consts with enums, used overloading on all functions/methods (I think) and since there's a function called Month and Weekday I renamed variables to _month and _weekday instead - not sure if that's proper or not.

thareh committed:
Renamed Weekday & Month functions & underscored variables

It's better this way imo
